Fix storybook knobs with timemachine test helper and config file

hotinglok opened the pull request:
Resolves #3896 

**Overall change:** 
Fixed storybook knobs as well as added `withTimeMachine` testhelper for future components which make use of `timemachine`.

**Code changes:**
- Added `time-machine.js` and `withTimeMachine.jsx` testhelper.
- Replaced `timemachine` in `/.storybook/config.js` with `startTimeMachine()`.
- Wrapped relevant components with `<WithTimeMachine>` from testhelper.

**EDIT: Can't seem to reply to comments for some reason, timestamp stories were removed by accident, should be back now.**

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

mykeels committed:
feat: use decorator with timemachine

hotinglok committed:
Removed eslint-disable in moment index.stories

hotinglok committed:
Added withTimemachine decorator relevant components

Added decorator to moment

Added decorator to RadioPageMain

Added decorator to RadioPageMain

hotinglok committed:
Removed unused import in config.js

mykeels committed:
fix: based on chromatic feedback

- Added the decorator to FrontPage stories
- Added to RadioPage stories
- Fixed the decorator in moment locales, so it’s only added once, rather than multiple times.

mykeels committed:
feat: start timemachine globally, and reset at every render

mykeels committed:
fix: lint issues

mykeels committed:
fix: freeze the second time ... aargh!

Renamed file fix

Fixed name change in RadioPage

hotinglok committed:
Linting, changed WithTimemachine to WithTimeMachine for consistency

Remove dangerouslySetInnerHTML from RadioPageBlocks

Refactor

mykeels committed:
fix: rename file for case sensitivity

mykeels committed:
fix: reset the timemachine, before setting it again

Correct brand images in three services

Correct og image for Nepali

mykeels committed:
fix: hail mary

mykeels committed:
fix: knobs for components that require timemachine

hotinglok committed:
Merge branch 'latest' into storybook-timemachine

thekp reviewed (commented):
Why are you deleting `ArticlesTimestamp.index.stories.jsx` ? 

edit: 

~nvm, I just read the thread on slack.
I think we shouldn't delete it and add `.addParameters({ chromatic: { disable: true } })` to it if we don't want it checked on chromatic.~
e.g: https://github.com/bbc/simorgh/blob/latest/src/app/containers/Brand/index.stories.jsx#L10

nvm me again, it looks like ArticlesTimestamp already had that ^ configured

hotinglok committed:
Restoring accidentally deleted file

hotinglok committed:
Merge branch 'latest' of github.com:bbc/simorgh into storybook-timemachine

hotinglok committed:
Merge branch 'storybook-timemachine' of github.com:bbc/simorgh into storybook-timemachine

oluoluoxenfree committed:
Merge branch 'latest' into storybook-timemachine

oluoluoxenfree commented:
@lokh01 how can we test this without the timestamp stories?

hotinglok commented:
@thekp @oluoluoxenfree added an edit in the original comment since I can't seem to reply to either of your comments. Removed by accident, added back now.

thekp commented on src/app/lib/utilities/moment/index.stories.jsx:
```diff
@@ -67,7 +68,6 @@ const OCTOBER = 1571097600000;
 const NOVEMBER = 1573776000000;
 const DECEMBER = 1576368000000;
 
-/* eslint-disable prettier/prettier */
```
repost: https://github.com/bbc/simorgh/pull/4178/files#r333233667

hotinglok replied:
Misread it first time round, fixed now.

hotinglok committed:
Added back eslint rule

HarveyPeachey reviewed (commented):
LGTM but might need some documentation or comments about what time machine does.

hotinglok committed:
Added comments to time-machine.js

hotinglok committed:
Merge branch 'latest' into storybook-timemachine

HarveyPeachey reviewed (approved):
LGTM

thekp reviewed (approved):
Nice work 💯 

but looks like you have some linting issues

hotinglok committed:
Linting

hotinglok committed:
Merge branch 'storybook-timemachine' of github.com:bbc/simorgh into storybook-timemachine

ryanmccombe commented on src/app/containers/RadioPageMain/index.stories.jsx:
```diff
@@ -2,10 +2,10 @@ import React from 'react';
 import { storiesOf } from '@storybook/react';
 import { inputProvider } from '@bbc/psammead-storybook-helpers';
 import { withKnobs } from '@storybook/addon-knobs';
+import WithTimeMachine from '../../../testHelpers/withTimeMachine';
```
Does the alias work here?
```suggestion
import WithTimeMachine from '#testHelpers/withTimeMachine';
```

FK78 replied:
Yes.

ryanmccombe commented on src/app/lib/utilities/moment/index.stories.jsx:
```diff
@@ -10,6 +10,7 @@ import {
   GEL_SPACING_DBL,
 } from '@bbc/gel-foundations/spacings';
 import { GEL_FF_REITH_SANS } from '@bbc/gel-foundations/typography';
+import WithTimeMachine from '../../../../testHelpers/withTimeMachine';
```
Does the alias work here?
```suggestion
import WithTimeMachine from '#testHelpers/withTimeMachine';
```

FK78 replied:
Yes.

ryanmccombe commented on src/testHelpers/withTimeMachine.jsx:
```diff
@@ -0,0 +1,25 @@
+  useEffect(() => {
+    resetTimeMachine();
+    startTimeMachine(true);
+    return () => {
```
Wrapper needed?
```suggestion
return resetTimeMachine;
```

lokh01 replied:
Functionality is still there without it, what does the wrapper do/improve on in this case?

hotinglok replied:
Functionality is there without it, what does the wrapper do/improve on in this case?

ryanmccombe replied:
I'm suggesting removing the wrapper you have, not adding another one :)

before:
```
return () => {
  resetTimeMachine ();
}
```
after:
```
return resetTimeMachine;
```

hotinglok replied:
I don't even have the excuse that it's friday :P, nice catch, my b

ryanmccombe replied:
👍 my two-word comment wasn't great

https://www.youtube.com/watch?v=_K-L9uhsBLM

FK78 committed:
Update src/app/lib/utilities/moment/index.stories.jsx

Co-Authored-By: Ryan McCombe <ryanmccombe@gmail.com>

hotinglok committed:
Replaced path to WithTimeMachine with alias

FK78 committed:
Update src/app/containers/RadioPageMain/index.stories.jsx

Co-Authored-By: Ryan McCombe <ryanmccombe@gmail.com>

hotinglok committed:
Merge branch 'storybook-timemachine' of github.com:bbc/simorgh into storybook-timemachine

hotinglok committed:
Removed unnecessary wrapper in withTimeMachine

hotinglok committed:
Merge branch 'latest' into storybook-timemachine

j-pendlebury reviewed (approved):
Looks good 👍

thekp commented on src/app/lib/utilities/moment/index.stories.jsx:
```diff
@@ -89,100 +90,664 @@ const editorialWhitelist = [
 ];
 
 const methods = [
-  { what: '[default format]', method: locale => moment(A_DATE).locale(locale).format() },
```
the formatting messed up again 😢

hotinglok committed:
Linting pls

thekp reviewed (approved):
Thanks for the fix 💯
